### PR TITLE
Drop UpdateGServer worker task if domain is blocked

### LINKED ISSUE #12601: UpdateGServer checks servers that are blocked.
- [x ] I have searched open and closed issues for duplicates

### Bug Description

UpdateGServer checks servers that are blocked.

The server `*.activitypub-troll.cf` is blocked on my instance, but still it is checked by UpdateGServer.

This is what the entry in the worker queue looks like:
`56497884 	UpdateGServer 	https://2cjt1orib.activitypub-troll.cf 	2023-01-02 18:59:16 	40`

### Steps to Reproduce

1.  Block a server
2.  Wait for UpdateGServer to be executed
3.  Check the worker queue

Actual Result:

The blocked server is checked

Expected Result:

The blocked server should not be checked.

### Platform Info

Friendica Version: Friendica 'Giant Rhubarb' 2023.03-dev - 1507

Friendica Source: git

PHP version: 8.1.

### PR DESCRIPTION
**Pull request by MrPetovan**
Fix #12601 

This PR also paves the way for a glorious future where GServer and Contact entities will have UriInterface properties 🤩

### CODE REVIEW

File: src/Worker/UpdateContact.php [hunk 1]
```diff
@@ -38,6 +39,11 @@ class UpdateContact
 	 */
 	public static function execute(int $contact_id)
 	{
+		// Silently dropping the task if the contact is blocked
+		if (Contact::isBlocked($contact_id)) {
```
Comment (annando) [inline src/Worker/UpdateContact.php:43]: Please also check if the contact's server could be blocked as well.

File: tests/src/Model/GServerTest.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
+<?php
```
Comment (annando) [inline tests/src/Model/GServerTest.php:1]: Think of adding the license header here.

File: src/Worker/UpdateContact.php (hunk 1 above)
Comment (MrPetovan) [reply src/Worker/UpdateContact.php:43]: It is already part of `Contact::isBlocked`.

File: tests/src/Model/GServerTest.php (hunk 2 above)
Comment (MrPetovan) [reply tests/src/Model/GServerTest.php:1]: Done.

**Pull request merged**